src/analysis.zig

```diff
@@ -285,7 +315,7 @@ pub fn formatFunction(
                                 if (is_comma or tag == .keyword_const) try writer.writeByte(' ');
                             }
                         } else {
-                            try writer.writeAll(offsets.nodeToSlice(tree.*, type_expr));
+                            try analyser.writeTypeFromTypeNode(.of(type_expr, handle), referenced, writer);
```

Comment:
formatting/stringifying a function should ideally not perform code analysis. Instead, `Analyser.Type.Data` should have a dedicated field for functions instead of the current `other`. This is more in line with how we represent other types like pointers or arrays.

This is change is going to require some more refactoring so I would suggest to leave that for a separate PR.

Reply:
Would it be okay to merge this PR first? I've found the improvements to hover to be really helpful